.circleci/config.yml

```diff
@@ -0,0 +1,49 @@
+version: 2
+jobs:
+  build:
+    working_directory: ~/micrometer
+    environment:
+      GRADLE_OPTS: '-Dorg.gradle.jvmargs="-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError"'
+    docker:
+      - image: circleci/openjdk:8-jdk
+    steps:
+
+      - checkout
+
+      - restore_cache:
+          key: gradle-dependencies-{{ checksum "build.gradle" }}
+
+      - run:
+          name: downloadDependencies
+          command: ./gradlew downloadDependencies --console=plain
+
+      - save_cache:
+          key: gradle-dependencies-{{ checksum "build.gradle" }}
+          paths:
+            - ~/.gradle
+
+      - run:
+          name: run tests
+          command: ./gradlew test
+
+      - run:
+          name: collect test reports
+          when: always
+          command: |
+            mkdir -p ~/micrometer/test-results/junit/
+            find . -type f -regex ".*/build/test-results/.*xml" -exec cp {} ~/micrometer/test-results/junit/ \;
```

Comment:
This is an existing issue so I don't think it needs to be solved in this pull request, but this will copy all of the test result XML files into a single folder. This would not be a problem if they were all uniquely named, but JUnit Jupiter test report files are all named `TEST-junit-jupiter.xml` so they overwrite each other in the destination folder, losing test results. I propose we change the logic to make a `test-results` directory somewhere and then make a directory under it for each sub project that has test results. All test results XML files for a sub project are put under its directory. Then CircleCI will list the results for each project, instead of just a generic `junit` like it does not. Also, none of the results would be lost. If my bash foo were up to par, I would have just written the code to do this...